annotation/tests/test_distribution.py

```diff
@@ -1277,3 +1286,192 @@ def test_get_page_number_combinations(
+                "validation": [
+                    {
+                        "task_id": 0,
+                        "file_id": 10,
+                        "pages": [1, 2, 3],
+                        "user_id": 1,
+                    },
+                ]
+            },  # Pages in work,
+            [
+                {
+                    "task_id": 1,
+                    "file_id": 10,
+                    "pages": [1, 2, 3, 4, 5, 6],
+                    "user_id": 2,
+                }
+            ],  # Tasks - Assuming an empty list is valid here
+            [
+                {
+                    "task_id": 1,
+                    "file_id": 10,
+                    "pages": [1, 2, 3, 4, 5, 6],
+                    "user_id": 2,
+                }
+            ],  # Expected output
+        ),
+        (
+            [{"file_id": 0, "pages_number": 6}],
+            [],
+            [User(user_id=2, default_load=10, overall_load=10)],
+            1,  # job_id, just a plain integer
+            ValidationSchema.hierarchical,
+            [{"file_id": 1}],  # Validation files to distribute
+            10,
+            {
+                "validation": [
+                    {
+                        "task_id": 0,
+                        "file_id": 10,
+                        "pages": [1, 2, 3],
+                        "user_id": 1,
+                    },
+                ]
+            },  # Pages in work,
+            [
+                {
+                    "task_id": 1,
+                    "file_id": 10,
+                    "pages": [1, 2, 3, 4, 5, 6],
+                    "user_id": 2,
+                }
+            ],  # Tasks - Assuming an empty list is valid here
+            [
+                {
+                    "task_id": 1,
+                    "file_id": 10,
+                    "pages": [1, 2, 3, 4, 5, 6],
+                    "user_id": 2,
+                }
+            ],  # Expected output
+        ),
+    ],
+)
+def test_distribute_main_script(
+    files: FilesForDistribution,
+    annotators: DistributionUser,
+    validators: DistributionUser,
+    job_id: int,
+    validation_type: ValidationSchema,
+    validation_files_to_distribute: FilesForDistribution,
+    extensive_coverage: int,
+    pages_in_work,
+    tasks,
+    expected_output,
+):
+    mock_db = MagicMock(spec=Session)
+    mock_db.flush.return_value = None
+
+    with patch(
+        "annotation.distribution.main.create_db_tasks", return_value=None
+    ), patch(
+        "annotation.distribution.main.distribute_tasks", return_value=tasks
+    ), patch(
+        "annotation.distribution.main.distribute_tasks_extensively",
+        return_value=tasks,
+    ), patch(
+        "annotation.distribution.main.remove_pages_in_work",
+        side_effect=lambda annotation_tasks, annotations_in_work: [
+            page
+            for page in annotation_tasks
+            if page not in annotations_in_work
+        ],
+    ):
```

Comment:
I've also changed so that instead of that side_effect function i have a lambda expression